api/types/change/types.proto

```diff
@@ -53,8 +55,13 @@ enum State {
     // COMPLETE indicates the phase is complete
     COMPLETE = 2;
 
+
+
+    // VALIDATION_FAILED indicates validation phase is failed
+    VALIDATION_FAILED = 3;
```

Comment:
I think we should add validation as a failure `Reason` rather than a different state

Reply:
@kuujo 
I understand your point and I would like to do that as well but the reason that I used a different state, its because we don't need to rollback a change when it fails because of validation. That means in the code where we are rolling back a failed change, we need to make sure it is not because of validation which means we should include reason in checking of the conditions. Do you think Is this OK?